lib/puppet/provider/exec/pwsh.rb

```diff
@@ -0,0 +1,63 @@
+require 'puppet/provider/exec'
+
+Puppet::Type.type(:exec).provide :pwsh, :parent => Puppet::Provider::Exec do
+  confine :operatingsystem  => [ :windows ]
+
+  commands :pwsh => 'pwsh.exe'
+
+  desc <<-EOT
+    Executes PowerShell Core commands. One of the `onlyif`, `unless`, or `creates`
+    parameters should be specified to ensure the command is idempotent.
+
+    Example:
+        # Rename the Guest account
+        exec { 'rename-guest':
+          command   => '$(Get-WMIObject Win32_UserAccount -Filter "Name=\'guest\'").Rename("new-guest")',
```

Comment:
This example needs to be changed. Use `Get-CimInstance` instead of `Get-WMIObject` as WMI is not present in pwsh.

Reply:
Fixed